enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,500 @@
+that are annotated in a particular way, so it could use the same logic to "delete" resources that are now
+matching a filter, just deleting the resources for the component isn't sufficient, as the component also
+needs to clean itself up in case it created any additional resources on the cluster or contributed any
+configuration.
+
+Therefore we plan to support turning a component on, but not turning it off.
+
+3. What are the implications for upgrades if a future upgrade would add a component or resource which would
+have been filtered out during install time?  
+
+There should be no implication here, the CVO has the list of annotations it will filter based on, if the
+new resources match those annotations, the new resources will also be filtered(never applied to the cluster).
+
+4. How prescriptive do we want to be about what can/can't be turned off?  Components need to opt into
+this by annotating their resources, so it's not completely arbitrary.
+
+This will need to be evaluated on a case by case basis as a component considers adding the annotation
+to its resources that will allow it to be filtered out/disabled.
+
+5. What if a user specifies a capability name to exclude that doesn't match anything?
+
+The installer won't know it's not going to match anything, so it should do nothing.  The CVO
+can tell it doesn't match anything, so it can provide a warning on the clusterversion resource, but
+it should not be an error since a user might put a keyword into the list in anticipation of upgrading
+to a new OCP version where the keyword will match (and thus exclude) a new capability, or perhaps
+the keyword used to match a capability that no longer exists after an upgrade.
+
+Along the same lines, if a user both "Excludes" and "Includes" a capability(if we ever add Include as
+a thing you can do), that should be treated as an error/invalid configuration.
+
+6. What to do(if anything) for components with interdependencies, to ensure a user doesn't break
+enabled components by disabling a dependency?  Options include:
+
+* Do nothing other than document dependencies so users know what not to turn off
+* Don't even annotate dependency resources for filtering, so if something is a dependency it cannot be turned off
+* Logic in the install or CVO that intelligently analyzes the filters the user has supplied and checks
+  for dependency issues (least desirable solution imho).
+
+Our current plan is to only annotate resources that are not depended upon (aka "leaf" components)
+
+### Open Questions
+
+1. What to do for components where disabling them has implications on other components or the way certain
+apis behave.  Example: disabling the internal registry changes the behavior of imagestreams
+(can't push to the imagestream anymore to push content to the internal registry) as well as the assumptions
+made by tools like new-app/new-build (create imagestreams that push to the internal registry).
```

Comment:
We should address this with documentation and good error messages that lead the user to enable the missing capability if some component doesn't work.